lib/livebook_web/live/output.ex

```diff
@@ -27,60 +29,85 @@ defmodule LivebookWeb.Output do
     """
   end
 
-  defp group_outputs(outputs) do
-    outputs = Enum.filter(outputs, &(&1 != :ignored))
-    group_outputs(outputs, [])
+  defp group_output_views(output_views) do
+    output_views = Enum.reject(output_views, &match?(%{output: :ignored}, &1))
+    group_output_views(output_views, [])
   end
 
-  defp group_outputs([], groups), do: groups
+  defp group_output_views([], groups), do: groups
 
-  defp group_outputs([output | outputs], []) do
-    group_outputs(outputs, [{[output], standalone?(output)}])
+  defp group_output_views([view | views], []) do
+    group_output_views(views, [{[view], standalone?(view.output)}])
   end
 
-  defp group_outputs([output | outputs], [{group_outputs, group_standalone?} | groups]) do
-    case standalone?(output) do
+  defp group_output_views([view | views], [{group_views, group_standalone?} | groups]) do
+    case standalone?(view.output) do
       ^group_standalone? ->
-        group_outputs(outputs, [{[output | group_outputs], group_standalone?} | groups])
+        group_output_views(views, [{[view | group_views], group_standalone?} | groups])
 
       standalone? ->
-        group_outputs(
-          outputs,
-          [{[output], standalone?}, {group_outputs, group_standalone?} | groups]
+        group_output_views(
+          views,
+          [{[view], standalone?}, {group_views, group_standalone?} | groups]
         )
     end
   end
 
   defp standalone?(:ignored), do: false
-  defp standalone?(text) when is_binary(text), do: false
+  defp standalone?({:stdout, _text}), do: false
   defp standalone?({:text, _text}), do: false
   defp standalone?({:error, _message, _type}), do: false
   defp standalone?(_output), do: true
 
-  defp composite?({:frame_dynamic, _}), do: true
+  defp composite?({:frame, _outputs, _info}), do: true
   defp composite?(_output), do: false
 
-  defp render_output(text, %{id: id}) when is_binary(text) do
-    # Captured output usually has a trailing newline that we can ignore,
-    # because each line is itself an HTML block anyway.
-    text = String.replace_suffix(text, "\n", "")
-    live_component(LivebookWeb.Output.TextComponent, id: id, content: text, follow: true)
+  defp render_output({:stdout, text}, %{id: id}) do
+    text = if(text == :__pruned__, do: nil, else: text)
```

Comment:
Nitpick: I would send the `:__pruned__` to the component, since the component needs to handle nil anyway. OR, replace `:__pruned__` by `nil` throughout. Or is there a reason why we need two values?